src/components/Composer/implementation/index.native.tsx

```diff
@@ -57,6 +57,22 @@ function Composer(
         inputCallbackRef(autoFocus ? textInput.current : null);
     }, [autoFocus, inputCallbackRef, autoFocusInputRef]);
 
+    useEffect(() => {
+        if (!textInput.current || !textInput.current.setSelection || !selection || isComposerFullSize) {
+            return;
+        }
+
+        // We are setting selection twice to trigger a scroll to the cursor on toggling to smaller composer size.
+        const timeoutID = setTimeout(() => {
+            textInput.current?.setSelection((selection.start || 1) - 1, selection.start);
+            textInput.current?.setSelection(selection.start, selection.start);
+        }, 0);
```

Comment:
Interesting, but your proposal did not mention anything about a `setTimeout()` before. This looks a bit hacky. @rayane-djouah what are your thoughts?

Reply:
BTW this setTimeout was added after the behavioural changes of IOS apps after the bridge less pr https://github.com/Expensify/App/pull/48160 I handled similar delay related IOS bug in https://github.com/Expensify/App/pull/51859 too.

Reply:
Confused about the state of that PR you linked what changes were added? It looks like a file was renamed and a comment updated 🙃

Reply:
Also I don't think that "because we did it before" is a good explanation for why it is acceptable. You had some alternate proposals IIRC why did we choose this route vs. another?

Reply:
And if you had been upfront about the proposal with a `setTimeout()` I likely would have pushed back here. Sure we can work something out though. Let's see what @rayane-djouah thinks.

Reply:
What I am trying to say is that setTimeout has nothing to do with my proposal. It was working perfect without it until the bridgeless [PR](https://github.com/Expensify/App/pull/48160) was merged, it caused a lot of IOS specific auto-focus issues too. Unfortuntely we didn't reach a correct root cause then, only proved as the cause after testing the reverting of the PR and the solution we used in all those cases was giving some delay, [here](https://github.com/Expensify/App/pull/51859) is one example and similarly delaying fixes our issue here too. I checked it again with lastest main merging if it would work without the delay but it doesn't 👍

Reply:
Do you have another example? The PR you linked doesn't really explain it for me.

>setTimeout has nothing to do with my proposal

I hear you, but either way, it's a problem. Is it possible to find a different workaround or address the root cause?

Reply:
The delay doesn't matter I think what it only needs to be called on the next rendering cycle so I chose the minimum possible delay. This is the only workaround I can think of, if you have any other approach please forward it or we can place some comment on this one and proceed with it 👍

Reply:
Discussing internally now.

Reply:
@WoLewicki what do you make of this conversation? 😄  I'd appreciate another opinion in here and seems you would have a good understanding of the [bridgeless changes](https://github.com/Expensify/App/pull/48160) and whether this is related or if there is any longer term solution to some of these focus workarounds.

Reply:
As mentioned by @FitseTLT, I researched it a bit and with bridgeless mode, there came a flag for batching all updates from event loop: https://github.com/facebook/react-native/blob/2e10ba945f6f2dca8d8cf1bb0db1d67fd50f2235/packages/react-native/src/private/featureflags/ReactNativeFeatureFlags.js#L172. What's more, this flag will no longer exist in 0.77: https://github.com/facebook/react-native/commit/2d9fe115789d1e00d2fee3440ec372025c7b2a4b#diff-413daa06293fe26f5032083b606e1a98042da0e71bc84cb2280aa39bbc09bc50R206 so we have really no option to switch it back to `false`.

To be honest, I would be up for fixing the current issues with `setTimeout` just to replicate the previous behavior. But on the long run, I'd go through those issues case by case, and see why is the `setTimeout` needed at all. 
For things like focus, we have to align sending the focus request with the native side, so the native part of the component is ready for it. Those issues probably come from dispatching the event with wrong timing and it worked only because the renders were dispatched later before.

For example, I fixed one of those places in https://github.com/Expensify/App/pull/48160/commits/52a976a3e1b99ee294fd73b9f856ecc64ca7dd79 since the library was dispatching the callback not when the native modal was hidden, but when the JS animation ended, which could be 2 different moments.

So, there is no easy solution for this, and we have to either replicate the previous behavior everywhere, or go through all the issues one by one and try to fix the root cause. The second option will for sure take much time and effort, so I'd start with the first one where it is needed now.

Reply:
Ok cool I'm good with either solution. But if we are going to add a `setTimeout()` then let's leave a good comment to explain exactly why it's needed in case these issues are resolved later.

Reply:
Perfect, thanks!